akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -364,6 +420,19 @@ private[remote] class Decoder(
         }
       }
 
+      private def applyIncomingInstruments(decoded: InboundEnvelope, headerBuilder: HeaderBuilder): Unit = {
```

Comment:
Good idea, moved it to Decoder.
And it's creating instruments per-materialization.